### PR TITLE
Add + character to allowed namespace element chars (AO-18082)

### PR DESCRIPTION
**Pull request by tomekkolo**

### CODE REVIEW

**Review by adamiklukasz — DISMISSED** [review]

File: v2/internal/util/metrictree/parser.go [hunk 1]
```diff
@@ -204,6 +204,9 @@ func isValidIdentifier(s string) bool {
 		case el >= '0' && el <= '9':
 		case el == '-' || el == '_':
 		case el == '.':
+		case el == '+':
+		case el == '&':
+		case el == '?':
```
Comment (lootek) [inline v2/internal/util/metrictree/parser.go:209]: please allow `=` as well so we can fully support URL query strings

File: v2/internal/util/metrictree/parser_test.go [hunk 2]
```diff
@@ -283,14 +306,14 @@ func TestParseNamespaceElement_InvalidScenarios(t *testing.T) {
...
 		" metr",
 		"metr ",
-		"asd+",
+		"asd;",
 		"{asd[}",
 		"[group=]",
 		"[=id]",
 		"[=]",
-		"[gr+]",
+		"[gr^]",
```
Comment (lootek) [inline v2/internal/util/metrictree/parser_test.go:314]: `^` has a special meaning for regexes, please use sth other than that

Comment (tomekkolo) [reply v2/internal/util/metrictree/parser_test.go:314]: this test checks invalid characters. ^ is used outside of regex, and it is forbidden

File: v2/internal/util/metrictree/parser.go (hunk 1 above)
Comment (tomekkolo) [reply v2/internal/util/metrictree/parser.go:209]: `=` is a separator actually, so it cannot be part of namespace element name

Comment (lootek) [reply v2/internal/util/metrictree/parser.go:209]: then there's no point for allowing & and ? - please drop them, we can always add them back if needed

File: v2/internal/util/metrictree/parser_test.go (hunk 2 above)
Comment (lootek) [reply v2/internal/util/metrictree/parser_test.go:314]: ah, ok, I was misled by square brackets 👍

File: v2/internal/util/metrictree/parser.go (hunk 1 above)
Comment (tomekkolo) [reply v2/internal/util/metrictree/parser.go]: done

**Review by lootek — APPROVED** [review]

**Review by adamiklukasz — APPROVED** [review]

**Pull request merged**